### DISCUSSION TITLE
sqlite and android?  missing libsqlite3.so.0

### DISCUSSION THREAD

**Discussion by bburdette** [tauri-apps/tauri#7340] [Q&A] [answered]
Just getting started with tauri.  I have an existing rust web server which uses rusqlite.  I'm able to run my app in desktop mode (cargo build and then invoke the app from the cli), but in android (cargo tauri android dev) it crashes with missing `libsqlite3.so.0`.  

I've heard that android typically has sqlite built in.  Is it possible to utilize the built in libsqlite of android?  Do I need to build my own sqlite lib for arm?  How do dependencies like this work in android development?  

I see there are sqlite plugins for tauri, but not sure if those solve my linkage problem, and anyway don't want to rewrite my app to use a different sqlite library if I can avoid it.

**Comment by FabianLars** [comment, marked as answer]
> I see there are sqlite plugins for tauri, but not sure if those solve my linkage problem, and anyway don't want to rewrite my app to use a different sqlite library if I can avoid it.

Yeah, don't use it. It doesn't make sense if you already have an implementation, especially since the plugin doesn't have a rust api. And it doesn't have explicit mobile support either (i'd imagine it fails with the same errors).

tbh i don't know how to really solve this in a way that uses the system's libsqlite3.so file, _if_ that's even possible and that may be a better question to ask the sqlite (bindings) maintainers.
But i'd recommend just statically linking it, libsqlite (statically linked) is really not large at all. To do this adding the `bundled` feature flag to the rusqlite dep should be enough, if you want to link it against the libsqlite library on the system look at the last points here: https://github.com/rusqlite/rusqlite#notes-on-building-rusqlite-and-libsqlite3-sys - tho i thought that this is tricky when targetting android so i thought you may already use the `bundled` feature flag in which case i have no idea what's going on...

> How do dependencies like this work in android development?

In context of _dynamic_ libraries i _think_ this is something we've not figured out / implemented yet either 🤔

**Reply by bburdette** [reply]
Ok nice!  Didn't occur to me they might have a bundle option.  No longer crashing from libsqlite - now I'm on to crashing on libssl and various other deps from my web server.  I'll see what I can do to bundle or eliminate those deps too.
